test/test_interpreter.cpp

```diff
@@ -48,3 +49,232 @@ TEST_SUITE("extract_filename")
+
+        REQUIRE(interp_ptr != nullptr);
+    }
+}
+
+TEST_SUITE("is_match_magics_manager")
+{
+    // This test case checks if the function `is_match` correctly identifies strings that match
+    // the regex pattern used in `xmagics_manager`. It sets up a scenario where strings that should
+    // match the pattern are passed and checks if the function returns true.
+    TEST_CASE("is_match_true")
+    {
+        xcpp::xmagics_manager manager;
+
+        bool result1 = manager.is_match("%%magic");
+        bool result2 = manager.is_match("%magic");
+
+        REQUIRE(result1 == true);
+        REQUIRE(result2 == true);
+    }
+
+    // This test case checks if the function `is_match` correctly identifies strings that do not match
+    // the regex pattern used in `xmagics_manager`. It sets up a scenario where strings that should not
+    // match the pattern are passed and checks if the function returns false.
+    TEST_CASE("is_match_false")
+    {
+        xcpp::xmagics_manager manager;
+
+        bool result1 = manager.is_match("not a magic");
+        bool result2 = manager.is_match("%%");
+        bool result3 = manager.is_match("%");
+
+        REQUIRE(result1 == false);
+        REQUIRE(result2 == false);
+        REQUIRE(result3 == false);
+    }
+}
+
+TEST_SUITE("clone_magics_manager")
+{
+    // This test case checks if the function `clone_magics_manager` returns a non-null pointer
+    // when called. It doesn't require any specific setup as it's testing the default behavior
+    // of the function, and checks if the function returns a non-null pointer.
+    TEST_CASE("clone_magics_manager_not_null")
+    {
+        xcpp::xmagics_manager manager;
+
+        xcpp::xpreamble* clone = manager.clone();
+
+        REQUIRE(clone != nullptr);
+    }
+
+    // This test case checks if the function `clone_magics_manager` returns a cloned object
+    // of the same type as the original. It calls the function and checks if the type of the
+    // returned object matches the type of the original `magics_manager`.
+    TEST_CASE("clone_magics_manager_same_type")
+    {
+        xcpp::xmagics_manager manager;
+
+        xcpp::xpreamble* clone = manager.clone();
+
+        REQUIRE(dynamic_cast<xcpp::xmagics_manager*>(clone) != nullptr);
+
+        delete clone;
+    }
+}
+
+TEST_SUITE("xpreamble_manager_operator")
+{
+    // This test case checks if the `xpreamble_manager` correctly registers and accesses
+    // a `xpreamble` object. It sets up a scenario where a `xpreamble` object is registered
+    // with a name and checks if the object can be accessed using the same name.
+    TEST_CASE("register_and_access")
+    {
+        std::string name = "test";
+        xcpp::xpreamble_manager manager;
+        xcpp::xmagics_manager* magics = new xcpp::xmagics_manager();
+        manager.register_preamble(name, magics);
+
+        xcpp::xholder_preamble& result = manager.operator[](name);
```

Comment:
We should probably annotate with `XEUS_CPP_API` the `xholder_preamble` class...

Reply:
Alright, I've added this

Reply:
Hey @vgvassilev, thank you so much for the help, it worked!